pkgs/os-specific/linux/trace-cmd/kernelshark.nix

```
@@ -1,22 +1,22 @@
-{ lib, mkDerivation, fetchzip, qtbase, cmake, asciidoc
+{ lib, stdenv, fetchzip, qtbase, qtscxml, cmake, asciidoc
 , docbook_xsl, json_c, mesa_glu, freeglut, trace-cmd, pkg-config
-, libtraceevent, libtracefs, freefont_ttf
+, libtraceevent, libtracefs, freefont_ttf, wrapQtAppsHook
 }:
 
-mkDerivation rec {
+stdenv.mkDerivation rec {
   pname = "kernelshark";
-  version = "2.2.1";
+  version = "2.3.0";
 
   src = fetchzip {
     url = "https://git.kernel.org/pub/scm/utils/trace-cmd/kernel-shark.git/snapshot/kernelshark-v${version}.tar.gz";
-    hash = "sha256-V25IzPDOt6V03wgIa/AJ0T8mRaGmXYuMCcvbSOKleY0=";
+    hash = "sha256-y4UpDcKnOgErgRTIYTWP6mFhdlsMP/Y214ZqCRxTAF0=";
   };
 
   outputs = [ "out" ];
 
-  nativeBuildInputs = [ pkg-config cmake ];
+  nativeBuildInputs = [ pkg-config cmake wrapQtAppsHook ];
 
-  buildInputs = [ qtbase json_c mesa_glu freeglut libtraceevent libtracefs trace-cmd ];
+  buildInputs = [ qtbase qtscxml json_c mesa_glu freeglut libtraceevent libtracefs trace-cmd ];
```

Qt 6 apps require qtwayland as a build input to support the wayland platform plugin.

```suggestion
  buildInputs = [ qtbase qtscxml qtwayland json_c mesa_glu freeglut libtraceevent libtracefs trace-cmd ];
```

---

This shouldn't be required as a buildInput and instead is a plugin that is required at runtime. AFAIR this is handled by `qt.enable=true` for NixOS, so I am not sure whether this is necessary.

---

When you add `qtwayland`, `wrapQtAppsHook` will just add it to `QT_PLUGIN_PATH` so that Qt can load it at runtime.

https://github.com/NixOS/nixpkgs/blob/bfb7a882678e518398ce9a31a881538679f6f092/nixos/modules/config/qt.nix doesn't have any mention of `wayland` :O

This also makes the app work better on systems that are not NixOS :D

---

Some more background on this:

- Qt 5's `wrapQtAppsHook` has `qtwayland` as a propagatedBuildInput for Linux packages. https://github.com/NixOS/nixpkgs/blob/880dae48fbbad1164f82e51879e83e96e90d7d24/pkgs/development/libraries/qt-5/5.15/default.nix#L371-L376
- Qt 6's `wrapQtAppsHook` does **not** propagate `qtwayland` (nor `qtbase`) anymore, which is why you need to add it to `buildInputs` yourself, if it's a GUI application that supports Wayland. https://github.com/NixOS/nixpkgs/blob/880dae48fbbad1164f82e51879e83e96e90d7d24/pkgs/development/libraries/qt-6/default.nix#L180-L186

IIRC, the reasoning for the change of `propagatedBuildInputs` was that Qt CLI apps shouldn't pull in Wayland.